javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -35,6 +39,19 @@
         return array;
     }
 
+    @SuppressWarnings("unchecked")
+    static <T> Object[] asArray(Iterable<? extends T> iterable) {
+        if (iterable instanceof Collection<?>) {
+            final Collection<? extends T> collection = (Collection<? extends T>) iterable;
+            return collection.toArray();
+        } else if (iterable instanceof Traversable) {
+            final Traversable<T> traversable = asTraversableAgain(iterable);
```

Comment:
The `iterable` that's returned by the test is not an instance of `Iterator`, it's some anonymous lambda.
I don't think we can consume it twice, I recommend reverting it to
```java
if (iterable instanceof Collection<?>) {
    final Collection<? extends T> collection = (Collection<? extends T>) iterable;
    return collection.toArray();
} else {
    final Traversable<T> traversable = asTraversableAgain(iterable);
    return asArray(traversable.iterator(), traversable.size());
}
```